### DISCUSSION TITLE
Add Livewire support

### DISCUSSION THREAD

**Discussion by fgschafer** [wintercms/winter#210] [Ideas]
I personally would love to be able tu use Livewire in WinterCMS.  IMO the most wanted feature to add (absolutely personal opinion).
There where some work in progress (at that moment for OC) on: https://github.com/LeMaX10/oc-livewire

**Comment by mjauvin** [comment]
That should work fine with Winter CMS by the way.

**Comment by bennothommo** [comment]
The AJAX framework and data attributes framework in Winter CMS already work like Livewire. Livewire would also introduce a requirement for Blade, if I'm not mistaken. Is there any feature in Livewire in particular that Winter CMS cannot do?

**Reply by fgschafer** [reply]
Even though I love WinterCMS, Livewire is much more powerful and feature rich than the current Ajax framework on Winter.  The most important differences I can think of now are:
* As you work directly with the DATA in Livewire, you don't have to specify what partial or dom elemenet to update, or refresh.  You get much easier reactivity.  If data is changed, Livewire reacts to refresh selected dom parts.  It's really like having VUE.js bundled in.  Please take a deep look at this item because I think it't the most important of the list by far.
* With Livewire you don't have to write a single line of code in vanilla javascript, for example to add keaboard events, or click away, etc... When you have to write custom vanilla js you are actually working with the DOM, but in Livewire you are working directly with the DATA.
* Custom traits and methods to add pagination, file uploads, etc.
* I think there are much more things to write, though I'm not an expert on the subject.

**Comment by Artful-dodger** [comment]
Has anyone got this to work?

**Reply by LukeTowers** [reply]
I'm not sure anyone's bothered to try tbh. Have you tried? Did you run into any issues?

**Reply by Artful-dodger** [reply]
I Have tried with October cms and wintercms, the plugin does not seem to register for some reason

**Reply by bennothommo** [reply]
@Artful-dodger does it throw any errors? Or does it just simply not work?

**Reply by Artful-dodger** [reply]
No errors just could not see the plugin in the backend even though I had installed it via composer and could see the folder in the plugin directory

**Reply by bennothommo** [reply]
@Artful-dodger I've assumed you've run `php artisan winter:up` after installing through Composer?

**Comment by Artful-dodger** [comment]
Hi Ben,
Yes I have tried php artisan winter:up but no luck but I do get an error “Call to undefined method Illuminate\Support\Facades\Cache::remember()” which I can’t figure out.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows
